django/contrib/admin/static/admin/js/change_form.js

```diff
@@ -2,8 +2,19 @@
 {
     const inputTags = ['BUTTON', 'INPUT', 'SELECT', 'TEXTAREA'];
     const modelName = document.getElementById('django-admin-form-add-constants').dataset.modelName;
+    const submitButtons = document.querySelectorAll('input[type=submit]');
+
     if (modelName) {
         const form = document.getElementById(modelName + '_form');
+
+        form.addEventListener('submit', function(event) {
+            event.preventDefault();
+            submitButtons.forEach(button => {
+                button.setAttribute('disabled', true);
```

Comment:
I don't know enough of the _details_ on this kind of thing (perhaps someone in @django/accessibility can enlighten me), but is this a scenario where we'd want to also set `aria-disabled=true`?

Reply:
As I understand it... `aria-disabled` doesn't affect the application logic at all, so you can still submit the form if it's present. The screen reader will just announce that it's "dimmed" or similar, but it's still interactive. On the other hand, `disabled` will make the screen reader skip the button entirely - the user wouldn't know the button exists - so there is no point using them both together.

I played around a bit with a toy example locally and it seems ok to me like this. If the request takes a really long time it could be confusing I suppose - but I think that's the same for any long form submission. Probably something can be done to alleviate this but I'm not sure it's worth spending the time on in this PR.